pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/BaseFilterOperator.java

```diff
@@ -56,4 +73,8 @@ public final BaseFilterBlock getNextBlock(BlockId blockId) {
   }
 
   public abstract BaseFilterBlock nextFilterBlock(BlockId blockId);
+
+  public boolean alwaysFalse() {
```

Comment:
Made alwaysFalse() abstract in base class, and sub classes now have their custom implementations.